lockfile/src/javascript.rs

```diff
@@ -499,23 +499,59 @@ mod tests {
 
     #[test]
     fn lock_parse_yarn_v1() {
-        for p in [
-            include_str!("../../tests/fixtures/yarn-v1.lock"),
-            include_str!("../../tests/fixtures/yarn-v1.trailing_newlines.lock"),
-        ] {
-            let pkgs = YarnLock.parse(p).unwrap();
+        let with_trailing_newlines = include_str!("../../tests/fixtures/yarn-v1.lock");
+        let without_trailing_newlines = format!("{}\n", with_trailing_newlines.trim_end());
 
-            assert_eq!(pkgs.len(), 17);
+        for p in [with_trailing_newlines, &without_trailing_newlines] {
+            let pkgs = YarnLock.parse(p).unwrap();
 
-            assert_eq!(pkgs[0].name, "@yarnpkg/lockfile");
-            assert_eq!(pkgs[0].version, PackageVersion::FirstParty("1.1.0".into()));
+            assert_eq!(pkgs.len(), 20);
 
-            assert_eq!(pkgs[3].name, "cliui");
-            assert_eq!(pkgs[3].version, PackageVersion::FirstParty("7.0.4".into()));
+            let expected_pkgs = [
```

Comment:
Hmm, I see that this appears to be a subset of the packages defined in the fixture. What's the reason for having more packages over there than we are asserting on here?

Reply:
Sounds good. That's kind of what I figured: make sure they don't fail to parse but don't really need explicit assertions for them.